support-frontend/assets/pages/contributions-landing/contributionsLandingActions.js

```diff
@@ -505,12 +525,14 @@ const onPaymentResult = (paymentResult: Promise<PaymentResult>, paymentAuthorisa
               stripeAccountForContributionType[state.page.form.contributionType],
             ));
           } else {
-            if (result.error === 'amazon_pay_fatal') {
-              dispatch(setAmazonPayFatalError);
-            }
-            if (result.error === 'amazon_pay_try_other_card') {
-              // Must re-render the wallet widget in order to display amazon's error message
-              dispatch(setAmazonPayWalletWidgetReady(false));
+            if (paymentAuthorisation.paymentMethod === 'AmazonPay') {
+              if (result.error === 'amazon_pay_try_other_card' || result.error === 'amazon_pay_try_again') {
```

Comment:
Both payment-api and support-workers now return these codes. Anything else can be considered 'fatal' and so we disable amazon pay on the client.